torch/csrc/jit/codegen/cuda/runtime/helpers.cu

```diff
@@ -620,3 +620,64 @@ __device__ __bfloat print_impl(const char* name, __bfloat value) {
 }
 
 #define print(...) print_impl(#__VA_ARGS__, (__VA_ARGS__))
+
+//! Utility to print out bank conflicts given the address from each thread.
+__device__ void checkBankConflict(
```

Comment:
I am wondering if this can be done at compile time. I think we can traverse the lowered kernel, find all unary set ops, and load-store ops with smem inputs/outputs. For each op, bind all its loop index and `blockIdx.x` to 0, and `threadIdx.x` to 0-31, and use the expression evaluator to evaluate the index.

Reply:
This way, we can write `TORCH_CHECK(fusion.bankConflict() == 0)` in unit tests.